### PR TITLE
SONARIAC-1118 S6470 should not raise an issue when COPY Instruction contains HereDoc

### PR DESCRIPTION
**Pull request by jonas-wielage-sonarsource**

### CODE REVIEW

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/TransferInstruction.java [hunk 1]
```diff
@@ -28,9 +28,19 @@
...
   List<Flag> options();
 
+  /**
+   * @return The list of sources of type {@link Argument} to transfer.
+   */
   List<Argument> srcs();
 
+  /**
+   * @return The destination of type {@link Argument} to transfer to.
+   */
```
Comment (jonas-wielage-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/TransferInstruction.java:44]: As a first step I changed the transfer instruction and updated javaDoc.
As the changes in transfer instruction weren't needed, I still will keep this javaDoc as it's already finished.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/DirectoryCopySourceCheck.java [hunk 2]
```diff
@@ -43,12 +45,12 @@ public class DirectoryCopySourceCheck implements IacCheck {
   @Override
   public void initialize(InitContext init) {
     init.register(AddInstruction.class, DirectoryCopySourceCheck::checkAdd);
-    init.register(CopyInstruction.class, DirectoryCopySourceCheck::checkCopy);
+    init.register(CopyInstruction.class, ignoringHeredoc(DirectoryCopySourceCheck::checkCopy));
```
Comment (mstachniuk) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/DirectoryCopySourceCheck.java:48]: Please add test case from ticket for this change

**Review by petertrr — APPROVED** [review]
LGTM, but please address Marcin's suggestion

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/DirectoryCopySourceCheck.java (hunk 2 above)
Comment (jonas-wielage-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/DirectoryCopySourceCheck.java:48]: Good catch!

**Pull request merged**